columnar/src/backend/columnar/columnar_customscan.c

```diff
@@ -302,6 +311,79 @@ columnar_customscan_init()
 	RegisterCustomScanMethods(&ColumnarScanScanMethods);
 }
 
+static bool IsCreateTableAs(const char *query)
+{
+	char *c, *t, *a;
+	char *haystack = (char *) palloc(strlen(query) + 1);
+	int16 i;
+
+	/* Create a lower case copy of the string. */
+	for (i = 0; i < strlen(query); i++)
+	{
+		haystack[i] = tolower(query[i]);
+	}
+
+	haystack[i] = '\0';
+
+	c = strstr(haystack, "create");
+	if (c == NULL)
+	{
+		pfree(haystack);
+		return false;
+	}
+
+	t = strstr(c + 6, "table");
+	if (t == NULL)
+	{
+		pfree(haystack);
+		return false;
+	}
+
+	a = strstr(t + 5, "as");
+	if (a == NULL)
+	{
+		pfree(haystack);
+		return false;
+	}
+
+	pfree(haystack);
+
+	return true;
+}
+
+static PlannedStmt *ColumnarPlannerHook(Query *parse, const char *query_string,
+										int cursorOptions, ParamListInfo boundParams)
+{
+	PlannedStmt *stmt;
+
+	if (PreviousPlannerHook)
+	{
+		stmt = (*PreviousPlannerHook)(parse, query_string, cursorOptions, boundParams);
+	}
+	else
+	{
+		stmt = standard_planner(parse, query_string, cursorOptions, boundParams);
+  }
+
+	/*
+	 * In the case of a CREATE TABLE AS query, we are not able to successfully
+	 * drop out of a parallel insert situation.  This checks for a CMD_SELECT
+	 * and in that case examines the query string to see if it matches the
+	 * pattern of a CREATE TABLE AS.  If so, set the parallelism to 0 (off).
+	 *
+	 * We are using strcasestr, which should give us
+	 */
+	if (parse->commandType == CMD_SELECT)
```

Comment:
It is fine to write into heap when parallel select will be planned. We should check if relation that is going to be written is columnar or something else.

Reply:
we only have access to what (if anything) is being written via the utility hook, that is fired off _after_ the data is written.  thus we cannot determine the type of table being written.  but, the side effect is that parallelism is turned off for a select into a write, which occurs in heap anyways.

Reply:
If we are disabling parallel CTAS and can't filter `into` table please put comment somewhere in code about this change.